### PR TITLE
Update vector.rst

### PR DESCRIPTION
**Pull request by eshavers**
add alternate methods for deleting vector attributes

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: docs/pyqgis_developer_cookbook/vector.rst [hunk 1]
```diff
@@ -561,6 +561,17 @@ For deletion of fields just provide a list of field indexes.
...
  if caps & QgsVectorDataProvider.DeleteAttributes:
      res = layer.dataProvider().deleteAttributes([0])
+     
+.. testcode:: vectors
+
+ #remove a single attribute column 
+ layer.dataProvider().deleteAttributes([0])
+
+ #remove multiple attributes with a list
+ lis= (1,2,3,4)
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/vector.rst:571]: We are getting a crash. Are you sure that `layer` has enough attributes. points.gml in the testsuite has only `fid`, `id` and `id2`...
```suggestion
 lis= (1,2)
```

**Comment by eshavers** [comment]
Ah, that makes sense. The suggested change looks fine.

**Comment by havatv** [comment]
> Ah, that makes sense. The suggested change looks fine.

@eshavers, it is possible for you to "accept" and commit suggestion here on github.com. Have a look at:
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request
If you do this in the "Files changed" tab, you can create a batch containing the suggestions you approve of, and then commit them in one go.

And thanks a lot for contributing to the QGIS documentation! Much appreciated.

File: docs/pyqgis_developer_cookbook/vector.rst [hunk 2]
```diff
@@ -561,6 +561,17 @@ For deletion of fields just provide a list of field indexes.
...
  if caps & QgsVectorDataProvider.DeleteAttributes:
      res = layer.dataProvider().deleteAttributes([0])
+     
+.. testcode:: vectors
+
+ #remove a single attribute column 
+ layer.dataProvider().deleteAttributes([0])
+
+ #remove multiple attributes with a list
+ lis = (1, 2)
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/vector.rst:571]:
```suggestion
 lis = [1, 2]
```
Maybe this will prevent the crash (Segmentation fault)?

Comment (DelazJ) [reply docs/pyqgis_developer_cookbook/vector.rst:571]: Maybe I miss something but if the layer has [3 fields (hence [0, 1, 2])](https://github.com/qgis/QGIS-Documentation/pull/5378#discussion_r418334836), and 0 has been deleted three lines above, then index 2 no longer exists.

File: docs/pyqgis_developer_cookbook/vector.rst [hunk 3]
```diff
@@ -561,6 +561,20 @@ For deletion of fields just provide a list of field indexes.
 
  if caps & QgsVectorDataProvider.DeleteAttributes:
      res = layer.dataProvider().deleteAttributes([0])
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/vector.rst:563]: @eshavers I've restarted the build a couple of times. No success. So Something might be buggy somewhere. But would this fix what you want to show (remove more than one field in a row)? Unless I misunderstand and you want to pass a named list as parameter?
```suggestion
     res = layer.dataProvider().deleteAttributes([0, 1])
```
Having to add fields that we delete later based on the reverse count of fields makes the sample code look quite complicated in its current shape to me and does not focus directly on what we want to do: "show users how to delete multiple fields in a row"

Comment (eshavers) [reply docs/pyqgis_developer_cookbook/vector.rst:563]: I know what you mean. Yet, to fit in the method without breaking later steps it seems necessary. It also may be useful to see the steps (create/delete) back to back

**Comment by havatv** [comment]
All checks passed!
Thanks, @eshavers!

File: docs/pyqgis_developer_cookbook/vector.rst (hunk 3 above)
Comment (DelazJ) [reply docs/pyqgis_developer_cookbook/vector.rst:563]: The create/delete process is shown in the previous example, isn't it? Or do I misunderstand u?

**Comment by DelazJ** [comment]
> All checks passed!

Should this be enough to merge PRs? I find the example complicate, particularly for beginners and quite repeating the one just above. Unless I miss something...

**Comment by DelazJ** [comment]
> Should/could this be backported to 3.10?

Technically they are compatible

**Comment by havatv** [comment]
> > All checks passed!
> 
> Should this be enough to merge PRs? I find the example complicate, particularly for beginners and quite repeating the one just above. Unless I miss something...

I agree that "All checks passed" is not enough to merge PRs. :-)

**Comment by eshavers** [comment]
I am new to all this so please excuse niece questions. Above, the update says the request was successfully merged and closed. Yet I do not see the edits in the master branch.

Regarding the usefulness of the addition, the benefit is two fold as I see it. 1. the command is run without the if statement (which is confusing if you find these instructions without reading all the previous setup, as I did). 2. the use of a list instead of only deleting individual fields. 

I think suggesting that it is too complicate may not give readers/user enough credit.

**Comment by havatv** [comment]
> I am new to all this so please excuse niece questions. Above, the update says the request was successfully merged and closed. Yet I do not see the edits in the master branch.

The edits are reflected in the master branch - have a look at
https://github.com/qgis/QGIS-Documentation/blob/master/docs/pyqgis_developer_cookbook/vector.rst
Thanks again!

**Comment by eshavers** [comment]
Indeed, thanks

**Pull request merged**